### PR TITLE
Add Ember.immediateObserver support

### LINKED ISSUE #8: Implement `immediateObserver`
removal: https://github.com/emberjs/ember.js/pull/15894

### PR DESCRIPTION
**Pull request by thoov**
Related: https://github.com/emberjs/ember.js/pull/15894

Closes #8

### CODE REVIEW

**Comment by thoov** [comment]
I still need to add Function.prototype support

**Comment by thoov** [comment]
Yes still need to add a test for this case

**Comment by thoov** [comment]
@rwjblue Should be good now.

**Comment by rwjblue** [comment]
Dang, I made a rebase conflict by merging the other PR.

**Comment by thoov** [comment]
@rwjblue Fixed. We are going to be hitting a bunch of conflicts as they all touch `config/environment.js` and most hit `index.js` 

¯\\\_(ツ)_/¯

should be good to go

File: addon/initializers/prototype-extend.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+import config from 'ember-get-config';
+
+export function initialize() {
+  const { EmberENV } = config;
+
+  if (EmberENV.EXTEND_PROTOTYPES && EmberENV.EXTEND_PROTOTYPES.Function === true && Ember.immediateObserver) {
+    Function.prototype.observesImmediately = Ember.immediateObserver;
```
Comment (rwjblue) [inline addon/initializers/prototype-extend.js]: Can we move this to a vendor file as well? Sorry for not noticing until the last minute :(

File: vendor/immediate-observer.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      _Ember.assert(
+        'Immediate observers must observe internal properties only, not properties on other objects.',
+        typeof arg !== 'string' || arg.indexOf('.') === -1
+      );
+    }
+
+    return _Ember.observer.apply(this, arguments);
+  }
+
+  const EXTEND_PROTOTYPES = JSON.parse(decodeURIComponent(document.getElementsByName('dummy/config/environment')[0].content)).EmberENV.EXTEND_PROTOTYPES;
```
Comment (thoov) [inline vendor/immediate-observer.js]: @rwjblue I thought that global.EmberEnv contains the merged configs. I see the correctly merged content inside of the meta tag but not of the global. Do you know of a way to grab this from within vendor?

Comment (thoov) [reply vendor/immediate-observer.js]: Disregard this. Issue was with https://github.com/ember-cli/ember-disable-prototype-extensions

**Comment by thoov** [comment]
@rwjblue This should be good now

**Pull request merged**